### ISSUE TITLE
The reason why using all the clipart data set when adapt from VOC to clipart?

### ISSUE THREAD

**Issue by kinredon** [VisionLearningGroup/DA_Detection#6]
In the paper, when conducting the experiment of adaptation from VOC to clipart, you said:

> All images were used for both training(without labels) and testing.

In the file `parser_func.py`:

```
 elif args.dataset == "clipart":
            args.imdb_name = "clipart_trainval"
            args.imdbval_name = "clipart_trainval"
            args.set_cfgs = ['ANCHOR_SCALES', '[8, 16, 32]', 'ANCHOR_RATIOS', '[0.5,1,2]', 'MAX_NUM_GT_BOXES',
                             '20']
```
You use same training set and test set.

So, the reason why using all the clipart data set when adapting from VOC to clipart?

**Comment by kailigo** [comment]
same question. Could the authors give some explanations? Thanks.

**Comment by DanZhang123** [comment]
Low result than the paper!
VOC->clipart 29.53, but the results of the paper is 38.1 
I use trainval_net_global_local.py gc=True lc=True , bs=1, epoch=20
I don't know if there is a problem with my dataset. please ask the download address of clipart dataset.

looking forward to your reply!

**Comment by kinredon** [comment]
Are the results evaluated by the model epoch = 20?
You should use the model of epoch 7.

**Comment by DanZhang123** [comment]
thanks, but we use  the model of epoch 7,the results is 35.28, also low than 38.1 reported in the paper . and I use 500 images(all in trainval datasets ) for training . and 500 images(all in test datasets) for test.
 The command for training the mode lare as follows:
 1, CUDA_VISIBLE_DEVICES=1 python trainval_net_global_local.py \
                    --dataset pascal_voc --dataset_t clipart --net res101 --lc --gc --ef\
                    --cuda
2, CUDA_VISIBLE_DEVICES=0 python trainval_net_global_local.py \
                    --dataset pascal_voc --dataset_t clipart --net res101 --lc --gc\
                    --cuda
I don't know what my problem is.  looking forward to your reply!

**Comment by DanZhang123** [comment]
> Are the results evaluated by the model epoch = 20? You should use the model of epoch 7.
> […](#)
> Sent from my iPhone
> On Oct 18, 2019, at 09:47, DanZhang123 ***@***.***> wrote: Low result than the paper! VOC->clipart 29.53, but the results of the paper is 38.1 I use trainval_net_global_local.py gc=True lc=True , bs=1, epoch=20 I don't know if there is a problem with my dataset. please ask the download address of clipart dataset. looking forward to your reply! — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

thanks, but we use the model of epoch 7,the results is 35.28, also low than 38.1 reported in the paper . and I use 500 images(all in trainval datasets ) for training . and 500 images(all in test datasets) for test.
The command for training the mode lare as follows:
1, CUDA_VISIBLE_DEVICES=1 python trainval_net_global_local.py
--dataset pascal_voc --dataset_t clipart --net res101 --lc --gc --ef
--cuda
2, CUDA_VISIBLE_DEVICES=0 python trainval_net_global_local.py
--dataset pascal_voc --dataset_t clipart --net res101 --lc --gc
--cuda
I don't know what my problem is. looking forward to your reply!

**Comment by kinredon** [comment]
The result(38.1) reported in the paper uses all the images(1000) to both train and test.

**Comment by DanZhang123** [comment]
Why is the clipart dataset both trained and tested in this article? what is meaning？




| |
张丹
邮箱：danzhang163com@163.com
|

签名由 网易邮箱大师 定制

On 11/08/2019 00:03, kinredon wrote:

The result(38.1) reported in the paper uses all the images(1000) to both train and test.

**Comment by kinredon** [comment]
That means the data for training and testing are the same, i.e. 1000 images in total.

**Comment by DanZhang123** [comment]
What I understand is that this does not meet the general requirements of experimental setup, training and testing should use different pictures.



| |
张丹
邮箱：danzhang163com@163.com
|

签名由 网易邮箱大师 定制

On 11/08/2019 00:39, kinredon wrote:

That means the data for training and testing are the same, i.e. 1000 images in total.

**Comment by kinredon** [comment]
Yeah, you got it.

**Comment by DanZhang123** [comment]
thanks！




| |
张丹
邮箱：danzhang163com@163.com
|

签名由 网易邮箱大师 定制

On 11/08/2019 00:48, kinredon wrote:

Yeah, you got it.

**Comment by DanZhang123** [comment]
> Yeah, you got it.

**Comment by DanZhang123** [comment]
Hi ,I recently ran the code again, but we use the model of epoch 7,the results is 35.2, also low than 38.1 reported in the paper . I use 1000 images for training . and also for test.(I found that the result I said before was 35.28, what I said before was 500 images for training and 500 images for testing, I think I misremembered)
The command for training the mode lare as follows:
1, CUDA_VISIBLE_DEVICES=1 python trainval_net_global_local.py
--dataset pascal_voc --dataset_t clipart --net res101 --lc --gc --ef
--cuda
2, CUDA_VISIBLE_DEVICES=0 python trainval_net_global_local.py
--dataset pascal_voc --dataset_t clipart --net res101 --lc --gc
--cuda
I don't know what my problem is. looking forward to your reply!  And have you ever run this code? What is the result

**Comment by DanZhang123** [comment]
Especially when I use the following command for training, and testing in epoch7 or nearby, the results are especially low! I don't know why.

**Comment by kinredon** [comment]
Do you used both pascal voc 07 and 12？

**Comment by DanZhang123** [comment]
thanks, I got it,  I only used voc07 before.

**Comment by DanZhang123** [comment]
> Do you used both pascal voc 07 and 12？
> […](#)
> Sent from my iPhone
> On Nov 13, 2019, at 15:54, DanZhang123 ***@***.***> wrote: Especially when I use the following command for training, and testing in epoch7 or nearby, the results are especially low! I don't know why. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Ｈｅｌｌｏ, when I train on cityscape-->foggy cityscape, I meet the problem of RPN regression box loss becoming NaN,　My cityscape and foggy cityscape datasets is directity from this link (https://github.com/tiancity-NJU/da-faster-rcnn-PyTorch). I don＇t know where I made mistakes. 
Have you ever run these two datasetes(cityscape-->foggy cityscape)
Looking forward to your reply, thank you very much!
